src/pages/gen2/build-a-backend/auth/add-social-provider/index.mdx

```diff
@@ -303,6 +303,26 @@ import { signInWithRedirect } from 'aws-amplify/auth';
   }
 ```
```

Comment:
This is an existing code example, but I don't think it's a good/correct example:

1. `signInWithRedirect` is an async API, but in a Web app, it's not really `await`-able as this API navigates away from the Web app
2. When the social sign-in completes, if `await signInWithRedirect` gets evaluated again, it will throw `USER_ALREADY_AUTHENTICATED_EXCEPTION` and the subsequent API call will not be executed at all

Please refer to [this section](https://docs.amplify.aws/javascript/build-a-backend/auth/add-social-provider/#add-custom-state) for a correct example.

Reply:
addressed in last commit